src/nvme/ioctl.c

```diff
@@ -1675,7 +1675,7 @@ static int nvme_set_var_size_tags(__u32 *cmd_dw2, __u32 *cmd_dw3, __u32 *cmd_dw1
 
 	switch (pif) {
 	case NVME_NVM_PIF_16B_GUARD:
-		cdw14 = be_reftag & 0xffffffff;
+		cdw14 = (be_reftag >> 32) & 0xffffffff;
```

Comment:
Yeah, this is a problem where the host system's 0xffffffff doesn't match the big-endian reference tag. Unfortunately I think the fix offered here would cause issues in big-endian environments by shifting the tag to 0.

Swapping to big-endian might need to happen after adjusting to fit the actual size/constraints of the PIF, for both the reference and storage tags.

Reply:
indeed, the cpu_to_be64 should be done after the mask operation.

Reply:
I just updated my PR for your comments. 

However, I believe we should be using cpu_to_be32 since the cdw sizes are in 32 bits. In case of using cpu_to_be64, even after the mask operation, the values will be cleared to 0.

Please let me know what you think.